### PR TITLE
[Look&Feel] Use semantic headers for page, modal, & flyouts across the board

### PR DESCRIPTION
**Pull request by danieldong51**
### Description

This PR updates headers to be within small OuiText in pages, modals, and flyouts across OSD experiences.

### Index Patterns
| Scope | Before (Inspect) | After (Inspect) | After |
| ----- | ----- | ----- | ----- |
| Create | Create Before | Create Post 1 | Create Post 2 |
| Page | Index Pattern Page Before | Index Pattern Page Post 1 | Index Pattern Page Post 2 |
| Table | Index Patterns Table Before |  Table Post 1 | Table Post 2 |

### Data Sources
| Scope | Before (Inspect) | After (Inspect) | After |
| ----- | ----- | ----- | ----- |
| Accelerate | Accelerate Data Before | Accelerate Data Post 1 | Accelerate Data Post 2 |
| Connection | Connection Before | Connection Post 1 | Connection Post 2 |
| Connection: Create | Create connection Before | Create connection Post 1 | Create connection Post 2 |
| Create Data Source | Create Before | Create Post 1 | Create Post 2 |
| Direct | Direct Before | Direct Post 1 | Direct Post 2 | 
| Home | Home Before | Home Post 1 | Home Post 2 |
| S3 | S3 Before Inspect | S3 Post Inspect | S3 Post 
| Prometheus: Configure | Prometheus Configure Before | Prometheus Configure Post 1 | Prometheus Configure Post 2 |
| Prometheus: Review | Prometheus Review Before | Prometheus Review Post 1 | Prometheus Review Post 2 |

### Saved Objects
| Scope | Before (Inspect) | After (Inspect) | After |
| ----- | ----- | ----- | ----- |
| Import | Import Before Inspect | Import Post Inspect | Import Post | 
| Export | Export Before | Export Post 1 | Export Post 2 |
| Flyout | Flyout Before | Flyout Post 1 | Flyout Post 2 |

### Other
| Scope | Before (Inspect) | After (Inspect) | After |
| ----- | ----- | ----- | ----- |
| Overview | Overview Before | Overview Post 1 | Overview Post 2 |
| Settings | Settings Before Inspect | Settings Post Inspect | Settings Post |
| Sample Data | Sample Data Before | Sample Data Post 1 | Sample Data Post 2 |

## Changelog
- Refactor: [Look&Feel] Use semantic headers for page, modal, & flyouts across the board

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_source_configuration/prometheus/configure_prometheus_data_source.tsx [hunk 1]
```diff
@@ -88,9 +87,9 @@ export const ConfigurePrometheusDatasourcePanel = (props: ConfigurePrometheusDat
   return (
     <div>
       <EuiPanel>
-        <EuiTitle>
-          <h4>{`Configure Prometheus data source`}</h4>
-        </EuiTitle>
+        <EuiText size="s">
+          <h1>{`Configure Prometheus data source`}</h1>
```
Comment (virajsanghvi) [inline src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_source_configuration/prometheus/configure_prometheus_data_source.tsx:91]: is this a page title?

**Review by virajsanghvi — COMMENTED** [review]
Think all my comments were basically validating that increasing sizes were because they were page headers but we weren't using h1s.

Also, if this is a page header, why is it an h2?
image

File: src/plugins/data_source_management/public/components/data_source_creation_panel/create_data_source_panel_header.tsx [hunk 2]
```diff
@@ -3,22 +3,22 @@
...
 import { FormattedMessage } from '@osd/i18n/react';
 
 export const CreateDataSourcePanelHeader: React.FC = () => {
   return (
     <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
       <EuiFlexItem grow={false}>
-        <EuiTitle>
-          <h2>
+        <EuiText size="s">
+          <h1>
```
[the remark this answers is not in the document]
Comment (danieldong51) [reply src/plugins/data_source_management/public/components/data_source_creation_panel/create_data_source_panel_header.tsx:15]: Yes, it's the create data source page : 
Screenshot 2024-08-05 at 2 52 10 PM
Screenshot 2024-08-05 at 2 52 38 PM

File: src/plugins/data_source_management/public/components/data_source_home_panel/data_source_page_header.tsx [hunk 3]
```diff
@@ -14,14 +14,14 @@ export const DataSourceHeader: React.FC<DataSourceHeaderProps> = () => {
   return (
     <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
       <EuiFlexItem grow={false}>
-        <EuiTitle>
-          <h2>
+        <EuiText size="s">
+          <h1>
```
[the remark this answers is not in the document]
Comment (danieldong51) [reply src/plugins/data_source_management/public/components/data_source_home_panel/data_source_page_header.tsx:18]: Yeah, also the title of Data Sources page
Screenshot 2024-08-05 at 2 53 23 PM

File: src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_source_configuration/prometheus/configure_prometheus_data_source.tsx (hunk 1 above)
Comment (danieldong51) [reply src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_source_configuration/prometheus/configure_prometheus_data_source.tsx:91]: It's the title of this page for creating a Prometheus data source: 
Screenshot 2024-08-05 at 2 53 47 PM

**Comment by danieldong51** [comment]
> Think all my comments were basically validating that increasing sizes were because they were page headers but we weren't using h1s.
> 
> Also, if this is a page header, why is it an h2? image

My bad, that was the old screenshot before I started updating the header tags. Updated with the correct screenshot now:
Screenshot 2024-08-05 at 2 58 24 PM

**Review by zhongnansu — APPROVED** [review]

**Review by virajsanghvi — APPROVED** [review]

**Pull request merged**